### ISSUE TITLE
Failed to save configuration: Error Domain=NEVPNErrorDomain Code=5 "IPC failed"

### ISSUE THREAD

**Issue by ApplantaDev** [lxdcn/NEPacketTunnelVPNDemo#1]
When I try to run the app, I get the following error:

Failed to save configuration: Error Domain=NEVPNErrorDomain Code=5 "IPC failed" UserInfo={NSLocalizedDescription=IPC failed}
Error Domain=NEVPNErrorDomain Code=5 "IPC failed" UserInfo={NSLocalizedDescription=IPC failed}
Go!
Error Domain=NEVPNErrorDomain Code=1 "(null)"

Do I have to request entitlements from apple?

**Comment by davlxd** [comment]
Oh yes, there are 2 things about NetworkExtension: you cannot run on simulator, and you need entitlements even you just debug on your iPhone without releasing.

BTW this project is unfinished and I'm kinda abandoned it because of my procrastination -_-||| . I will finish this (with an elaborate installation guide) if you really need it :)

**Comment by ApplantaDev** [comment]
Thanks for the info. I was able to get the entitlements set up. Did you ever establish a VPN connection using the current code that you have now?

**Comment by davlxd** [comment]
Yes, I just complete this repo, and it works fine on my iPhone, with [server side](https://github.com/lxdcn/simple-vpn-demo) setup.

**Comment by ApplantaDev** [comment]
I just followed the instructions from both repos. I am still getting this as thew output in the console


VPN Status changed:
Invliad
VPN Status changed:
Disconnected...
Save successfully

VPN status is always invalid

**Comment by davlxd** [comment]
My bad, my instructions are not elaborate enough. Here it is.
  - After build & run `simple-vpn-demo`, your terminal should be like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/simple-vpn-demo-terminal-before.png).
  - After build & run `NEPacketTunnelVPNDemo` on your iPhone, Xcode log window should look like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/xcode-log-window-before.png), iPhone screen should look like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/iphone-before.png). And this is where you are.
  - The `Connect` at the center of iPhone screen is a **BUTTON**, after you click it, your iPhone screen should look like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/iphone-after.png), and Xcode log window should look like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/xcode-log-window-after.png). `simple-vpn-demo` terminal window should look like [this](https://raw.githubusercontent.com/lxdcn/NEPacketTunnelVPNDemo/master/doc/simple-vpn-demo-terminal-after.png), which means VPN server is now receiving traffic.
  - Your IP address of [http://whatismyipaddress.com/](http://whatismyipaddress.com/) on your iPhone should be different before and after click the `Connect` button.

Q.E.D.

**Comment by davlxd** [comment]
Hi Touchcore, does this work for you? I am closing this issue if it does.

**Comment by ApplantaDev** [comment]
I will try this today and get back to you. Thanks a lot!

**Comment by ApplantaDev** [comment]
I got it working. However, the following method is never called in PacketTunnelProvider.swift:
startTunnel(options: [String : NSObject]?, completionHandler: @escaping (Error?) -> Void) 
Do you have any idea?

**Comment by davlxd** [comment]
To verify it's called, you might need to set a break point and debug after Xcode attach to `NEPacketTunnelVPNDemoTunnel`, like [this](http://stackoverflow.com/questions/24031612/how-to-debug-ios-8-extensions-with-nslog/24288906#24288906), kind of cumbersome.

**Comment by ApplantaDev** [comment]
For some reason I can establish a vpn connection but no methods in the PacketTunnel Provider file are ever called. This is my stackoverflow link http://stackoverflow.com/questions/42701530/packettunnelprovider-network-extension-not-called-swift-3

**Comment by visionstech** [comment]
@lxdcn 
I just followed the instructions from both repos. I am still getting this as thew output in the console
NEPacketTunnelVPNDemo[9687:1248755] refreshPreferences: HangTracerEnabled: 0
2018-02-27 11:52:43.975500+0530 NEPacketTunnelVPNDemo[9687:1248755] refreshPreferences: HangTracerDuration: 500
2018-02-27 11:52:43.975565+0530 NEPacketTunnelVPNDemo[9687:1248755] refreshPreferences: ActivationLoggingEnabled: 0 ActivationLoggingTaskedOffByDA:0
VPN Status changed:
Invliad
VPN Status changed:
Disconnected...
Save successfully
Go!
VPN Status changed:
Connecting...
VPN Status changed:
Disconnected...

**Comment by kukushechkin** [comment]
@visionstech were you able to overcome this Connecting-Disconnected problem?

**Comment by TheMrugraj** [comment]
Any update on this issue ?

**Comment by ms-tii** [comment]
> Any update on this issue ?

Hi Sir, Do you able to solve the Connecting-Disconnected problem?

**Comment by TheMrugraj** [comment]
Yes, In my case I was missing an additional Capability for `Personal VPN` to add in entitlements.
